From 1cc37758aa83c23c4278b033bf9cc5084b6b711c Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Wed, 7 Dec 2022 01:48:56 +0100
Subject: [PATCH 6/7] RunGrepTest: Windows does not have a /bin/echo

Therefore, let's hard-code the absolute path to the MSYS2 "echo.exe" in
`$MSYSROOT/usr/bin/echo.exe`.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 RunGrepTest | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/RunGrepTest b/RunGrepTest
index d1ddf257..20ad33a5 100755
--- a/RunGrepTest
+++ b/RunGrepTest
@@ -1062,16 +1062,16 @@ fi
 if $valgrind $vjs $pcre2grep --help | $valgrind $vjs $pcre2grep -q 'callout scripts in patterns are supported'; then
   echo "Testing pcre2grep script callouts"
   echo "--- Test 1 ---" >testtrygrep
-  $valgrind $vjs $pcre2grep '(T)(..(.))(?C"/bin/echo|Arg1: [$1] [$2] [$3]|Arg2: $|${1}$| ($4) ($14) ($0)")()' $srcdir/testdata/grepinputv >>testtrygrep
+  $valgrind $vjs $pcre2grep '(T)(..(.))(?C"echo|Arg1: [$1] [$2] [$3]|Arg2: $|${1}$| ($4) ($14) ($0)")()' $srcdir/testdata/grepinputv >>testtrygrep
   echo "RC=$?" >>testtrygrep
   echo "--- Test 2 ---" >>testtrygrep
-  $valgrind $vjs $pcre2grep '(T)(..(.))()()()()()()()(..)(?C"/bin/echo|Arg1: [$11] [${11}]")' $srcdir/testdata/grepinputv >>testtrygrep
+  $valgrind $vjs $pcre2grep '(T)(..(.))()()()()()()()(..)(?C"echo|Arg1: [$11] [${11}]")' $srcdir/testdata/grepinputv >>testtrygrep
   echo "RC=$?" >>testtrygrep
   echo "--- Test 3 ---" >>testtrygrep
   $valgrind $vjs $pcre2grep '(T)(?C"|$0:$1$n")' $srcdir/testdata/grepinputv >>testtrygrep
   echo "RC=$?" >>testtrygrep
   echo "--- Test 4 ---" >>testtrygrep
-  $valgrind $vjs $pcre2grep '(T)(?C"/bin/echo|$0:$1$n")' $srcdir/testdata/grepinputv >>testtrygrep
+  $valgrind $vjs $pcre2grep '(T)(?C"echo|$0:$1$n")' $srcdir/testdata/grepinputv >>testtrygrep
   echo "RC=$?" >>testtrygrep
   echo "--- Test 5 ---" >>testtrygrep
   $valgrind $vjs $pcre2grep '(T)(?C"|$1$n")(*F)' $srcdir/testdata/grepinputv >>testtrygrep
@@ -1097,7 +1097,7 @@ if $valgrind $vjs $pcre2grep --help | $valgrind $vjs $pcre2grep -q 'callout scri
     $valgrind $vjs $pcre2grep -u '(T)(?C"|$0:$x{a6}$n")' $srcdir/testdata/grepinputv >>testtrygrep
     echo "RC=$?" >>testtrygrep
     echo "--- Test 2 ---" >>testtrygrep
-    $valgrind $vjs $pcre2grep -u '(T)(?C"/bin/echo|$0:$x{a6}$n")' $srcdir/testdata/grepinputv >>testtrygrep
+    $valgrind $vjs $pcre2grep -u '(T)(?C"echo|$0:$x{a6}$n")' $srcdir/testdata/grepinputv >>testtrygrep
     echo "RC=$?" >>testtrygrep
 
     if [ $nonfork = 1 ] ; then
-- 
2.48.1.windows.1

